src/rpc/blockchain.cpp

```diff
@@ -1532,6 +1537,63 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
     return ret;
 }
 
+UniValue requestblocks(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 3)
+        throw std::runtime_error(
+            "requestblocks (add|flush|status) ([\"hash_0\", \"hash_1\", ...])\n"
+            "\nPriorize blocks downloads.\n"
+            "\nArguments:\n"
+            "1. action            (string, required) the action to execute\n"
+            "                                        add  = add new blocks to the priority download\n"
+            "                                        flush = flush the queue (blocks in-flight will still be downloaded)\n"
+            "                                        status = get info about the queue\n"
+            "2. blockhashes       (array, optional) the hashes of the blocks to download\n"
+            "\nResult:\n"
+            "   add: <null>\n"
+            "   flush: <true|false> (if the the queue wasn't empty)\n"
+            "   status: [\"<blockhash in queue>\", \"<blockhash in queue>\" ...]\n"
+            "\nExamples:\n"
+            + HelpExampleCli("requestblocks", "add, \"'[\"<blockhash>\"]'\"")
+            + HelpExampleRpc("requestblocks", "add, \"'[\"<blockhash>\"]'\"")
+            );
+
+    if (request.params[0].get_str() == "flush") {
+        return UniValue(FlushPriorityDownloads());
+    }
+    else if (request.params[0].get_str() == "status") {
+        UniValue ret(UniValue::VARR);
+        std::map<uint256, const CBlockIndex*> prioBlocks = GetPriorityDownloads();
+        for(auto& kv: prioBlocks) {
+            ret.push_back(kv.first.ToString());
+        }
+        return ret;
+    }
+    else if (request.params[0].get_str() == "add") {
+        if (request.params.size() < 2) {
```

Comment:
In commit "[RPC] Add requestblocks - a simple way to priorize block downloads"

Don't really need this check, because the get_array call below will trigger it's own error about the value not being an array. But if you prefer this error, you should change the check to request.params[1].isNull() to avoid making a distinction between a null and a missing value (#10281, #10783).